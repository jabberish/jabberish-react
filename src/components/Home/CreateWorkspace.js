import React from 'react';
import PropTypes from 'prop-types';

import { Grid, makeStyles, Typography, Button } from '@material-ui/core';

const useStyles = makeStyles(() => ({
  button: {
    height: 140,
    width: 300,
    textDecoration: 'none'
  }
}));

const CreateWorkspace = ({ handleOpenDialog }) => {
  const classes = useStyles();
  return (
    <Grid item>
      <Button 
        className={classes.button} 
        onClick={handleOpenDialog} 
        variant="outlined" 
        color="primary"
      >
        <Typography variant="subtitle1" className={classes.typography}>
        Create Workspace
        </Typography>
      </Button>
    </Grid>
  );
};

CreateWorkspace.propTypes = {
  handleOpenDialog: PropTypes.func.isRequired
};

export default CreateWorkspace;
