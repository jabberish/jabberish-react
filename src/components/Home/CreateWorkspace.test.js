import React from 'react';
import { shallow } from 'enzyme';
import CreateWorkspace from './CreateWorkspace';

describe('CreateWorkspace component', () => {
  it('renders CreateWorkspace', () => {
    const wrapper = shallow(<CreateWorkspace handleOpenDialog={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
