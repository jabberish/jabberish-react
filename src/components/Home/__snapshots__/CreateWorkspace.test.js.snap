// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateWorkspace component renders CreateWorkspace 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-button-1"
    color="primary"
    onClick={[Function]}
    variant="outlined"
  >
    <WithStyles(ForwardRef(Typography))
      variant="subtitle1"
    >
      Create Workspace
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Button))>
</WithStyles(ForwardRef(Grid))>
`;
